Sources/RemoteMessaging/RemoteMessagingStore.swift

```diff
@@ -105,9 +109,18 @@ public final class RemoteMessagingStore: RemoteMessagingStoring {
 
         let context = database.makeContext(concurrencyType: .privateQueueConcurrencyType, name: Constants.privateContextName)
 
-        invalidateRemoteMessagingConfigs(in: context)
-        deleteScheduledRemoteMessages(in: context)
-        notificationCenter.post(name: Notifications.remoteMessagesDidChange, object: nil)
+        context.performAndWait {
```

Comment:
Just wrapping old code into a context.performAndWait and adding save at the end.